### PR TITLE
Add button component

### PR DESCRIPTION
**Pull request by JakeMaldonado**

### CODE REVIEW

File: src/components/Button.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+  return [...classes, ...baseClasses(underlineOnHover, bold), ...dynamicClasses(size, isDisabled, fullWidthMobile)].join(' ')
+}
+
+const Button = ({ text, disabled, kind, size, underlineOnHover, bold, onClick, type, fullWidthMobile, isRunning }) => {
+  kind = kind || 'primary'
+  size = size || 'base'
+
+  underlineOnHover = underlineOnHover || true
+  bold = bold || true
```
Comment (simpsonjoshua) [inline src/components/Button.js]: We can add default values when we destructure the values. For example:

`const Button = ({ kind = 'primary', size = 'base'})`

File: src/components/Button.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    if (kind === 'basic') {
+      return basicButtonClasses(underlineOnHover, bold, kind, size)
+    } else {
+      return buttonClasses(kind, underlineOnHover, bold, size, isDisabled, fullWidthMobile)
+    }
+  }
+
+  return(
+    <button disabled={ isDisabled(disabled, isRunning) } onClick={ onClick } type={ type } className={ classes(isDisabled, fullWidthMobile) } >
+      <div id="ui-button-content" >
```
Comment (simpsonjoshua) [inline src/components/Button.js]: Why is this `id` needed?

File: src/components/Button.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    'tw-font-bold',
+    'tw-leading-normal',
+    'tw-border-2',
+    `tag-button-${kind}`
+  ];
+
+  return [...classes, ...baseClasses(underlineOnHover, bold), ...dynamicClasses(size, isDisabled, fullWidthMobile)].join(' ')
+}
+
+const Button = ({ text, disabled, kind, size, underlineOnHover, bold, onClick, type, fullWidthMobile, isRunning }) => {
```
Comment (simpsonjoshua) [inline src/components/Button.js]: Can we start using `PropTypes` on all our components?

File: src/components/Button.js (hunk 1 above)
Comment (JakeMaldonado) [reply src/components/Button.js]: Good idea

File: src/components/Button.js (hunk 2 above)
Comment (JakeMaldonado) [reply src/components/Button.js]: Not needed! Forgot to remove when i copied the element over from tag

Comment (JakeMaldonado) [reply src/components/Button.js]: @simpsonjoshua hi 👋

Comment (simpsonjoshua) [reply src/components/Button.js]: @JakeMaldonado LOL 👋

Comment (simpsonjoshua) [reply src/components/Button.js]: @JakeMaldonado fusaros?

Comment (JakeMaldonado) [reply src/components/Button.js]: @simpsonjoshua LOL im down man

**Pull request merged**